hr_holidays_credit/tests/test_hr_holidays_credit.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (C) 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo.exceptions import ValidationError
+from odoo.tests import common
+
+_logger = logging.getLogger(__name__)
+
+
+class TestHrHolidaysCredit(common.TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.Employee = cls.env["hr.employee"]
+        cls.SudoEmployee = cls.Employee.sudo()
+        cls.Department = cls.env["hr.department"]
+        cls.SudoDepartment = cls.Department.sudo()
+
+        cls.LeaveType = cls.env["hr.leave.type"]
+        cls.SudoLeaveType = cls.LeaveType.sudo()
+        cls.Leave = cls.env["hr.leave"]
+        cls.SudoLeave = cls.Leave.sudo()
+        cls.Allocation = cls.env["hr.leave.allocation"]
+        cls.SudoAllocation = cls.Allocation.sudo()
+
+    def test_1(self):
```

Comment:
"""
        Test that creating a leave without allowing credit raises a ValidationError,
        and succeeds after enabling credit on the leave type.
        """
```suggestion
    def test_1(self):
```